src/freenet/client/filter/CSSTokenizerFilter.java

```diff
@@ -1801,27 +1802,23 @@ else if(HTMLelement.indexOf('#')!=-1)
 		}
 
 		if(attSelections!=null) {
-		    String[] attSelectionParts;
-
-		    for(String attSelection : attSelections) {
-		        if(attSelection.indexOf("|=")!=-1) {
-		            attSelectionParts=new String[2];
-		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("|="));
-		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("|=")+2,
-		                    attSelection.length());
-		        } else if(attSelection.indexOf("~=")!=-1) {
-		            attSelectionParts=new String[2];
-		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("~="));
-		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("~=")+2,
-		                    attSelection.length());
-		        } else if(attSelection.indexOf('=') != -1){
-		            attSelectionParts=new String[2];
-		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf('='));
-		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf('=')+1,
-		                    attSelection.length());
-		        } else {
-		            attSelectionParts=new String[] { attSelection };
-		        }
+
+			for(String attSelection : attSelections) {
+
+				// set the default
+				String[] attSelectionParts = new String[] { attSelection };;
+
+				List<String> operators = Arrays.asList("|=", "~=", "^=", "$=", "*=", "=");
+				for (String comparisonOperator : operators) {
+						if (attSelection.contains(comparisonOperator)) {
+							attSelectionParts = new String[2];
+							attSelectionParts[0] = attSelection
+									.substring(0, attSelection.indexOf(comparisonOperator));
+							attSelectionParts[1] = attSelection
+									.substring(attSelection.indexOf(comparisonOperator) + comparisonOperator.length());
+							break;
+						}
+					}
```

Comment:
The formatting and indentation of this whole thing is a bit wild, and there’s a stray `;` in line 1809. 🙂

Reply:
The indentation there follows the rules, but other parts partly break that.

Since we have no other CSS PRs right now, I tabified the file now.